### DISCUSSION TITLE
Is there a way to view the events stream in a ascending order by time?

### DISCUSSION THREAD

**Discussion by mingchau** [datalust/seq-tickets#1601] [Help]
I think displaying messages sorting in ascending order is more intuitive. Our IDE console and log files append the latest log messages to the end. It is also more convenient when it is necessary to analyze the context of the log. Thanks in advance.

![image](https://user-images.githubusercontent.com/18564039/177310683-1c4e33e0-274a-45a9-b3fb-0dfb9edc0b11.png)

**Comment by nblumhardt** [comment]
Thanks for your message, @mingchau.

There isn't any way to do this today; the UI is fundamentally organized around the newest-first layout, and while it would be nice to enable reading in the opposite direction as well, it's unlikely we'll add this in the near future.

When logging one large event in multiple lines as above, trying to combine all details of the event into a single log event - e.g. by storing query parameters and headers in structured event properties - will provide a much better viewing and searching experience, if you're able to make that change.

Best regards,
Nick

**Comment by denovellis** [comment]
I agree with @mingchau. I could be great if we can choose the default order in the settings.

**Comment by ste-phil-c4b** [comment]
I would also love to see this feature

**Comment by The-Futurist** [comment]
I recently submitted this discussion: https://github.com/datalust/seq-tickets/discussions/2498

Surely this is a matter of how the window pane itself behaves not Seq as such. Writing a new line by scrolling the screen down and writing the new line at the top is just one way for a text panel to behave. The ask here is to write the line at the next position available and if there's no more room then scroll the window pane text up one line and then write the new line at the very bottom of the window. 

That's all that the Windows console does, the app that's writing has no idea how the console reacts to new input, it just has a behavior. 

The Console policy is scroll window upwards (if there's no space for the next line)  write line at bottom, whereas Seq today is scroll window downwards write line at top. 

The Console is how almost all such things are handled, imagine typing commands and see all the previous typed stuff scrolled down and your next command appearing always at the top!

If that Seq log window is home-grown I imagine it would be a reasonably doable change for a javascript guru.
